### PR TITLE
Documentation Enanchement

### PR DESCRIPTION
**Pull request by nicolocarpignoli**
I totally re-organized the Readme.
I think that, for now, working on an AR.js Wiki is secondary. First thing user will see and one of the main reasons they will try AR.js is a comprehensive Readme, where it can be easy to find every important information.

Also, I think is best to put effort on a tiny self-contained doc on the Readme, at least, to start, and not using the separate project of AR.js doc.

So, the changelog is easy:

* re-organized sections: first thing user will see, after the "introduction" to the project, is a 'marker based' explanation and a "Get Started" section, which contains the "AR in 10 lines of HTML" example plus guides for beginners and advanced users. I linked medium articles and others.
* after that, comes the examples. I added @stemkoski fantastic work on AR.js examples here
* a section about tools (marker and arcode generator)
* then comes all other parts that I considered less important for an user who see to Readme for the first time
* "Last but not least" section containing separate sections of documentation - i.e. "How to release" is now in a separate file, because (as It is written :) ) is more for AR.js author than for users.
* Minor fixes such as grammar errors and typos.

Enjoy

### CODE REVIEW

**Comment by nicolocarpignoli** [comment]
"Conflicting files" is obviously misleading as I modify only one file, the Readme.md, that has to be completely replaced by the new one.

File: README.md [hunk 1]
```diff
@@ -38,47 +37,11 @@ to your screen.
...
-- **three.js** for being a great library to do 3d on the web.
-- **artoolkit**! years of development and experiences on doing augmented reality
-- **emscripten and asm.js**! thus we could compile artoolkit c into javascript
-- **chromium**! thanks for being so fast!
-
-Only thanks to all of them, I could do my part : Optimizing performance from 5fps on high-end
-phone, to 60fps on 2years old phone.
-
-After all this work done by a lot of people, we have a *web-based augmented reality solution fast enough for mobile*!
+# What means "Marker based"
```
Comment (ShawnOakley) [inline README.md]: This probably should read 'What "Marker based" means'

Comment (nicolocarpignoli) [reply README.md]: changed :)

**Comment by nicolocarpignoli** [comment]
As I tried to ping collaborators and had no luck, I will merge this as I already got good feedback on this PR and it has also been reviewed.

**Pull request merged**